### PR TITLE
Contrib Trim Galore

### PR DESCRIPTION
**Pull request by tloizou**
Contributing Trim Galore, a wrapper tool around Cutadapt and FastQC to consistently apply quality and adapter trimming to FastQ files, with some extra functionality for MspI-digested RRBS-type.

edit: now requires #1982

### CODE REVIEW

**Comment by boegel** [comment]
Jenkins: ok to test

**Comment by tloizou** [comment]
Dependencies added. Please test.

File: easybuild/easyconfigs/t/Trim_Galore/Trim_Galore-0.3.7.eb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+toolchain = {'name': 'dummy', 'version': 'dummy'}
+
+sources = ['%(namelower)s_v%(version)s.zip']
+source_urls = ['http://www.bioinformatics.babraham.ac.uk/projects/%(namelower)s']
+
+install_cmd = "cp -a %(namelower)s_zip/* %(installdir)s && chmod +x %(installdir)s/%(namelower)s"
+
+dependencies = [
+     ('FastQC', '0.10.1', '-Java-1.7.0_21'),
+     ('cutadapt', '1.5', '-goolf-1.4.10-Python-2.7.3'),
```
Comment (boegel) [inline easybuild/easyconfigs/t/Trim_Galore/Trim_Galore-0.3.7.eb]: if you include a dependency that uses a non-dummy toolchain, please make this easyconfig use that same toolchain, to make it clear

Comment (tloizou) [inline easybuild/easyconfigs/t/Trim_Galore/Trim_Galore-0.3.7.eb]: I have made the necessary changes. I have built FastQC with the goolf toolchain and put Java as a dependency and also built Trim Galore with goolf.

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by boegel** [comment]
@wpoely86: please review

File: easybuild/easyconfigs/c/cutadapt/cutadapt-1.5-goolf-1.4.10-Python-2.7.3.eb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+description = """ cutadapt removes adapter sequences
+ from high-throughput sequencing data. This is usually
+ necessary when the read length of the sequencing machine
+ is longer than the molecule that is sequenced, for
+ example when sequencing microRNAs.  """
+
+toolchain = {'name': 'goolf', 'version': '1.4.10'}
+
+source_urls = ['http://cutadapt.googlecode.com/files/']
+sources = ['%(namelower)s-%(version)s.tar.gz']
```
Comment (wpoely86) [inline easybuild/easyconfigs/c/cutadapt/cutadapt-1.5-goolf-1.4.10-Python-2.7.3.eb]: Use `SOURCELOWER_TAR_GZ` ?

File: easybuild/easyconfigs/c/cutadapt/cutadapt-1.5-goolf-1.4.10-Python-2.7.3.eb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+pyshortver = '.'.join(pyver.split('.')[:2])
+
+versionsuffix = "-%s-%s" % (python, pyver)
+
+dependencies = [
+    (python, pyver),
+]
+
+sanity_check_paths = {
+    'files': ['bin/cutadapt','lib/python2.7/site-packages/cutadapt/_align.so',],
```
Comment (wpoely86) [inline easybuild/easyconfigs/c/cutadapt/cutadapt-1.5-goolf-1.4.10-Python-2.7.3.eb]: Do: 
```python
pyshortver = '.'.join(pyver.split('.')[:2])
sanity_check_paths = {
    'files': [lib/python%s/site-packages/cutadapt/_align.so' % pyshortver],
}
```

**Comment by wpoely86** [comment]
small remarks, looks fine for the rest.

**Comment by tloizou** [comment]
Changes have been made. Please test.

**Comment by wpoely86** [comment]
lgtm

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/0a663aa8ffad5362160d for a full test report.

**Comment by boegel** [comment]
@tloizou: hmm, very old Python and Java here... any particular reason?

would you mind updating to Python 2.7.10 and Java 1.7.0_80?

we can add the Python 2.7.10 easyconfig for goolf/1.4.10 in another PR (just let me know)

**Comment by tloizou** [comment]
@boegel: There is no specific reason for the Python and Java versions...It's just what most of our users are still using. 

I can update to Python 2.7.10 and Java 1.7.0_80.

Do you want me to create a PR for Python 2.7.10-goolf-1.4.10?

**Comment by boegel** [comment]
@tloizou: let me tackle the PR for Python 2.7.10-goolf-1.4.10

**Comment by tloizou** [comment]
@boegel: Ok, so should I just update the versions in the config files?

**Comment by boegel** [comment]
@tloizou: yep, I'll make sure the Python PR is merged first so Jenkins is happy

**Comment by boegel** [comment]
Python 2.7.10 w/ goolf/1.4.10 in #1982

File: easybuild/easyconfigs/f/FastQC/FastQC-0.10.1-goolf-1.4.10.eb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+easyblock = "PackedBinary"
+
+name = 'FastQC'
+version = '0.10.1'
+
+homepage = 'http://www.bioinformatics.babraham.ac.uk/projects/fastqc/'
+description = """FastQC is a quality control application for high throughput sequence data.
+It reads in sequence data in a variety of formats and can either provide an interactive
+application to review the results of several different QC checks, or create an HTML based
+report which can be integrated into a pipeline."""
```
Comment (fgeorgatos) [inline easybuild/easyconfigs/f/FastQC/FastQC-0.10.1-goolf-1.4.10.eb]: LGTM! just leave a 1 space indent, for readability..

File: easybuild/easyconfigs/t/Trim_Galore/Trim_Galore-0.3.7-goolf-1.4.10.eb [hunk 5]
```diff
@@ -0,0 +1,28 @@
+easyblock = "PackedBinary"
+
+name = 'Trim_Galore'
+version = '0.3.7'
+
+homepage = 'http://www.bioinformatics.babraham.ac.uk/projects/trim_galore/'
+description = """A wrapper tool around Cutadapt and FastQC to consistently apply 
+quality and adapter trimming to FastQ files, with some extra functionality for 
+MspI-digested RRBS-type (Reduced Representation Bisufite-Seq) libraries."""
```
Comment (fgeorgatos) [inline easybuild/easyconfigs/t/Trim_Galore/Trim_Galore-0.3.7-goolf-1.4.10.eb]: ditto

**Comment by boegel** [comment]
hmm still conflicts, dafuq...

**Comment by boegel** [comment]
@wpoely86: can you try and get this synced with `develop` and resolve the conflicts? I gave it a try, but something is still wrong here...

**Comment by wpoely86** [comment]
@boegel done, see #1984

**Comment by boegel** [comment]
merged via #1984, thanks @tloizou and @wpoely86!

**Pull request merged**